### PR TITLE
Riemann fit patatrack

### PR DESCRIPTION
**Pull request by rovere**
Working implementation with RiemanFit. It also includes a minimalistic working example of running Eigen on GPU.

### CODE REVIEW

File: RecoPixelVertexing/PixelTrackFitting/test/BuildFile.xml [hunk 1]
```diff
@@ -14,6 +14,7 @@
 <use name="cuda"/>
 <use name="cuda-api-wrappers"/>
 <bin file="testEigenGPU.cu" name="testEigenGPU_t">
+  <flags CUDA_FLAGS="--expt-relaxed-constexpr -fmad=false -ftz=false -prec-div=true -prec-sqrt=true"/>
```
Comment (fwyzard) [inline RecoPixelVertexing/PixelTrackFitting/test/BuildFile.xml]: why `-fmad=false` ?

Comment (rovere) [reply RecoPixelVertexing/PixelTrackFitting/test/BuildFile.xml]: That was a list of flags to try and have a simpler assert in the code, but apparently, they did not work and I implemented the assert that went into the code. I can remove them.

File: RecoPixelVertexing/PixelTrackFitting/test/BuildFile.xml [hunk 2]
```diff
@@ -11,6 +11,15 @@
 <use   name="RecoTracker/TkTrackingRegions"/>
 <use   name="RecoPixelVertexing/PixelTriplets"/>
 <use   name="RecoPixelVertexing/PixelTrackFitting"/>
+<use name="cuda"/>
+<use name="cuda-api-wrappers"/>
```
Comment (makortel) [inline RecoPixelVertexing/PixelTrackFitting/test/BuildFile.xml]: Could these be moved inside the `<bin></bin>` block? IIUC then they are applied only to the `testEigenGPU_t` and not everything (ok, doesn't matter too much in practice yet, but less to clean up later).

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by fwyzard** [comment]
@cmsbot please test

**Comment by fwyzard** [comment]
please test

**Comment by smuzaffar** [comment]
let me check why cms-bot is not responding

**Comment by smuzaffar** [comment]
@fwyzard , can you please check https://github.com/cms-patatrack/cmssw/settings/hooks and see if  webhook issue/comment is being triggered and if you get any error back from cms-bot in response to that hook?

**Comment by fwyzard** [comment]
@smuzaffar for whatever reason the webhooks are not there any more ?
Let me try adding them back...

**Comment by fwyzard** [comment]
OK, found them... I have the webhooks for the whole "cms-patatrack" organisation, not in the "cmssw" repository.

The most recent ones I see are from few minutes ago: 
```
eec40060-1ba0-11e8-880e-d6c1c53e7284                2018-02-27 10:31:04
a3932680-1b9f-11e8-906e-ee7819bd0301                2018-02-27 10:21:49
83840ee0-1b9f-11e8-9838-4c7c80c74a43                2018-02-27 10:20:55
28f5af60-1b9f-11e8-9955-31d70c795411                2018-02-27 10:18:23
c1e8ead0-1b9e-11e8-8e94-f84fa19656a1                2018-02-27 10:15:30
b84da880-1b9e-11e8-98c0-bb85be98475a                2018-02-27 10:15:14
694ff2b0-1b9e-11e8-82ea-0bab66fc82f7                2018-02-27 10:13:02
```

**Comment by fwyzard** [comment]
And for all issues, the response is 200, and the message is
```
False issue_comment
Invalid or wrong signatures
```

**Comment by smuzaffar** [comment]
ok thanks, let me check

**Comment by smuzaffar** [comment]
ping

**Comment by smuzaffar** [comment]
@fwyzard , sorry my bad. There was a typo which is fixed now.

**Comment by fwyzard** [comment]
OK, thanks.

By the way, should we as @cmsbuild or @cmsbot to run the tests ?

**Comment by smuzaffar** [comment]
use @cmsbuild (which is optional) as mentioned in http://cms-sw.github.io/cms-bot-cmssw-cmds.html

**Pull request closed without merging**